python/plugins/processing/algs/qgis/RandomPointsOnLines.py

```diff
@@ -0,0 +1,257 @@
+        self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT,
+                                                              self.tr('Input layer'),
+                                                              [QgsProcessing.TypeVectorLine]))
+        self.addParameter(QgsProcessingParameterNumber(self.POINTS_NUMBER,
+                                                       self.tr('Number of points on each feature'),
+                                                       QgsProcessingParameterNumber.Integer,
+                                                       1, False, 1, 1000000000))
+        self.addParameter(QgsProcessingParameterDistance(self.MIN_DISTANCE,
+                                                         self.tr('Minimum distance between points'),
+                                                         0, self.INPUT, False, 0, 1000000000))
+        self.addParameter(QgsProcessingParameterNumber(self.MAXTRIESPERPOINT,
+                                                       self.tr('Maximum number of attempts per point'),
+                                                       QgsProcessingParameterNumber.Integer,
+                                                       10, False, 1, 1000))
+        self.addParameter(QgsProcessingParameterNumber(self.RANDOM_SEED,
+                                                       self.tr('Integer seed (for random)'),
+                                                       QgsProcessingParameterNumber.Integer,
+                                                       None, True, 1, 1000000000))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT,
+                                                            self.tr('Random points'),
+                                                            type=QgsProcessing.TypeVectorPoint))
+        self.addOutput(QgsProcessingOutputNumber(self.OUTPUT_POINTS,
+                                                 self.tr('Number of point generated')
+                                                 ))
+
+    def processAlgorithm(self, parameters, context, feedback):
+        source = self.parameterAsSource(parameters, self.INPUT, context)
+        if source is None:
+            raise QgsProcessingException(self.invalidSourceError(parameters,
+                                                                 self.INPUT))
+
+        pointCount = self.parameterAsInt(parameters, self.POINTS_NUMBER,
+                                         context)
+        minDistance = self.parameterAsDouble(parameters, self.MIN_DISTANCE,
+                                             context)
+        randSeed = self.parameterAsInt(parameters, self.RANDOM_SEED, context)
+        MaxTriesPerPoint = self.parameterAsInt(parameters,
+                                               self.MAXTRIESPERPOINT,
+                                               context)
+        fields = QgsFields()
+        fields.append(QgsField('id', QVariant.Int, '', 10, 0))
```

Comment:
Instead of this, I'd rather keep all the input fields intact and just append a new "point_num" field on the end. That'll be more useful for users.

Reply:
I will look into that. Perhaps tomorrow.

Reply:
I added this as an option. The default is not to include the attributes from the line.

Reply:
Can I ask you to revisit this? Making it optional deviates from the behaviour of all other similar algorithms, such as "points along geometry". We should always keep the input attributes intact and not make this optional.